lib/promoted/ruby/client.rb

```diff
@@ -69,21 +69,31 @@ def initialize (params={})
           @http_client = FaradayHTTPClient.new
           @validator = Promoted::Ruby::Client::Validator.new
 
-          # Thread pool to process delivery of shadow traffic. Will silently drop excess requests beyond the queue
-          # size, and silently eat errors on the background threads.
-          @pool = Concurrent::ThreadPoolExecutor.new(
-            min_threads: 0,
-            max_threads: 10,
-            max_queue: 100,
-            fallback_policy: :discard
-          )
+          @async_shadow_traffic = true
+          if params[:async_shadow_traffic] != nil
```

Comment:
I don't know Ruby.  Would it work to replace the whole block with the following?

```
@async_shadow_traffic = params[:async_shadow_traffic] || false
```

Reply:
That's a good pattern for false defaults, but I think in this case we want it to be true.